### ISSUE TITLE
Emitter particleConstructor type definition

### ISSUE THREAD

**Issue by tarling** [pixijs/pixi-particles#43]
Looking at the [example](https://github.com/pixijs/pixi-particles/blob/0d3d210cf1c3dd9ab733a835d4fabf6cf7dc42ed/docs/examples/js/ParticleExample.js#L124) and the [implementation](https://github.com/pixijs/pixi-particles/blob/master/src/Emitter.js#L484), I wonder if the definition here for`particleConstructor` should be 

```
public particleConstructor:typeof Particle;
```

instead of

```
public particleConstructor:(p:Particle, emitPosX:number, emitPosY:number, i:number)=>void;
```

When I try and use particleConstructor as per the example, it raises a type error

```
Type 'typeof PathParticle' provides no match for the signature '(p: Particle, emitPosX: number, emitPosY: number, i: number): void''
```

If you would be happy with this, let me know and I'll raise a PR

**Comment by andrewstart** [comment]
Whoops, looks like I went a little wild with the find & replace, and used the definition of a private method. While `typeof Particle` might be valid, I think that for constructors specifically, `new (emitter:Emitter)=>Particle` would be the syntax. I've never tried `typeof X` with a subclass of `X`, and have no idea if it would work.

**Comment by andrewstart** [comment]
This should be fixed with v2.1.1.

**Comment by tarling** [comment]
Thanks Andrew - you beat me to it
